src/utils/testsToStatusUpdate.ts

```diff
@@ -0,0 +1,49 @@
+import type { API } from "@storybook/manager-api";
+import type { StoryId } from "@storybook/types";
+
+import type { TestFieldsFragment, TestStatus, ViewportInfo } from "../gql/graphql";
+
+export type StatusUpdate = Parameters<API["experimental_updateStatus"]>[1];
+type StoryStatus = StatusUpdate[any]["status"];
+
+const statusMap: Record<TestStatus, StoryStatus> = {
+  IN_PROGRESS: "pending",
+  PASSED: null,
+  ACCEPTED: null,
+  PENDING: "warn",
+  FAILED: "error",
+  DENIED: "error",
+  BROKEN: "error",
+};
+
+// Just the bits we need
+type ReducedTest = Pick<TestFieldsFragment, "story" | "status">;
+
+const statusOrder: StoryStatus[] = ["unknown", "pending", "success", "warn", "error"];
+function chooseWorseStatus(status: StoryStatus, oldStatus: StoryStatus | null) {
+  if (!oldStatus) return status;
+
+  return statusOrder[Math.max(statusOrder.indexOf(status), statusOrder.indexOf(oldStatus))];
+}
```

Comment:
Hmm probably not great to re-implement this, But I don't really have a great place put this so i can export it in storybook..

Reply:
Yes, I had the same thought process...